test/benchgc/src/benchgc/pattern/base.py

```diff
@@ -0,0 +1,43 @@
+################################################################################
+# Copyright (C) 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+# SPDX-License-Identifier: Apache-2.0
+################################################################################
+
+import argparse
+from abc import ABC, abstractmethod
+
+from gc_mlir import ir
+
+
+class Pattern(ABC):
```

Comment:
Using OOP style may not be a right choice, I think. Please check how single op validation does?

Reply:
any reason for not using the OOP style?

Reply:
Because I did not see any chance of code reuse in this scenario. If you insist, then let's keep it this way and resolve the conversation.